### PR TITLE
Implement Dynamic Search Functionality for Search Bar

### LINKED ISSUE #247: Making Search Bar Responsive
The search bar should be responsive so that whatever is search using that a response is generated.
Please assign this to me @dohinafs under GSSOC 24 Extended .

### PR DESCRIPTION
**Pull request by Akshita246**
I implement a dynamic search functionality for the menu items of the website. This feature allows users to easily search and filter ice cream flavors by typing in the search bar. As they type, the page automatically updates to show only the matching items, improving user experience and making it more convenient for customers to find their desired flavors.
Changes Made:
Implemented JavaScript functionality to update the displayed menu items based on the search input in real-time.
Improved the user interface to provide a more seamless and intuitive search experience.
Related Issues:
Closes #247 .
Before:
ArcticDelights - Google Chrome 06-10-2024 20_16_34
After:
ArcticDelights - Google Chrome 06-10-2024 20_04_34

### CODE REVIEW

**Comment by Akshita246** [comment]
Hi @dohinafs ,

I hope you're doing well! I have submitted this PR as part of Hacktoberfest and GSSOC, and if you find it helpful, could you please add the `hacktoberfest-accepted` label?

Thank you for your time and consideration!

**Comment by Altair-05** [comment]
image
@Akshita246, the changes you made are not appearing here.  please check and update accordingly?

**Comment by Altair-05** [comment]
@Akshita246  Also can you tell me  issue number of this pr?

**Comment by Akshita246** [comment]
its #247

**Comment by Akshita246** [comment]
Hello i just checked and found out that the filter part of the code is not
responsive though it is not working. As my pull request #247 was for the
search bar which is present in the nav bar , I am going to raise another
issue for the filter part  so please assign me that under gssoc and
hacktoberfest

**Comment by Akshita246** [comment]
Hi this is to inform you that someone changed the some of the template and
cursor setting due to which the wishlist , cart and the search bar and
filter functionality which I have changed is not working please make sure
that if someone is changing anything it should not affect the existing
functionality of other

File: index.html [hunk 1]
```diff
@@ -906,6 +906,8 @@ <h4>- Stacy Stone</h4>
     <script src="https://unpkg.com/aos@2.3.1/dist/aos.js"></script>
     <script>
         AOS.init();
+      </script>   
+=======
```
Comment (Altair-05) [inline index.html:910]: why u used script tags without any use?

File: script.js [hunk 2]
```diff
@@ -466,8 +466,25 @@ document.addEventListener('DOMContentLoaded', () => {
...
+document.getElementById('search-box').addEventListener('input', function () {
+    const searchTerm = this.value.toLowerCase();
+    const menuItems = document.querySelectorAll('.menu .box');
+
+    menuItems.forEach(item => {
+        const itemName = item.querySelector('h3').textContent.toLowerCase();
+        
+        // Check if the search term is in the item name
+        if (itemName.includes(searchTerm)) {
+            item.style.display = '';  // Show the item if it matches
```
Comment (Altair-05) [inline script.js:479]: Items are not displayed after clicking the search button. Please review this function

File: index.html [hunk 3]
```diff
@@ -906,6 +906,8 @@ <h4>- Stacy Stone</h4>
     <script src="https://unpkg.com/aos@2.3.1/dist/aos.js"></script>
     <script>
         AOS.init();
+      </script>
```
Comment (Altair-05) [inline index.html:909]: Why did you use script tags without any functionality?

**Review by Altair-05 — CHANGES_REQUESTED** [review]

File: script.js (hunk 2 above)
Comment (Akshita246) [reply script.js:479]: Ok Let me be very clear the person who change the syling of the site disabled the functionalities of even the wishlist, add to cart and the search bar  and even the button of add to cart is not working so now i have to review all the code again give me some time

File: index.html (hunk 3 above)
Comment (Akshita246) [reply index.html:950]: index html - basic-icecream-website - Visual Studio Code 09-10-2024 19_31_04
but there is no such code

File: script.js (hunk 2 above)
Comment (Altair-05) [reply script.js:479]: sure

Comment (Akshita246) [reply script.js:522]: I have updated the code you can now check the search icon and the filter functionalities .

**Comment by Akshita246** [comment]
I have resolved the issue of  the code of the filter and the search icon functionalities you can check and if there is no more problem please let me know so that i can close the pull request.

**Comment by dohinafs** [comment]
Kindly resolve the conflicts

**Comment by Akshita246** [comment]
> Kindly resolve the conflicts

which???

**Pull request closed without merging**